radarr/movieeditor.go

```diff
@@ -13,17 +13,36 @@ const bpMovieEditor = bpMovie + "/editor"
 
 // BulkEdit is the input for the bulk movie editor endpoint.
 // You may use starr.True(), starr.False(), starr.Int64(), and starr.String() to add data to the struct members.
+// Use Availability.Ptr() to add a value to minimum availability, and starr.ApplyTags.Ptr() for apply tags.
 type BulkEdit struct {
-	MovieIDs            []int64 `json:"movieIds"`
-	Monitored           *bool   `json:"monitored,omitempty"`
-	QualityProfileID    *int64  `json:"qualityProfileId,omitempty"`
-	MinimumAvailability *string `json:"minimumAvailability,omitempty"` // tba
-	RootFolderPath      *string `json:"rootFolderPath,omitempty"`      // path
-	Tags                []int   `json:"tags,omitempty"`                // [0]
-	ApplyTags           *string `json:"applyTags,omitempty"`           // add
-	MoveFiles           *bool   `json:"moveFiles,omitempty"`
-	DeleteFiles         *bool   `json:"deleteFiles,omitempty"`        // delete only
-	AddImportExclusion  *bool   `json:"addImportExclusion,omitempty"` // delete only
+	MovieIDs            []int64          `json:"movieIds"`
+	Monitored           *bool            `json:"monitored,omitempty"`
+	QualityProfileID    *int64           `json:"qualityProfileId,omitempty"`
+	MinimumAvailability *Availability    `json:"minimumAvailability,omitempty"` // tba
+	RootFolderPath      *string          `json:"rootFolderPath,omitempty"`      // path
+	Tags                []int            `json:"tags,omitempty"`                // [0]
+	ApplyTags           *starr.ApplyTags `json:"applyTags,omitempty"`           // add
+	MoveFiles           *bool            `json:"moveFiles,omitempty"`
+	DeleteFiles         *bool            `json:"deleteFiles,omitempty"`        // delete only
+	AddImportExclusion  *bool            `json:"addImportExclusion,omitempty"` // delete only
+}
+
+// Availability is an enum used as MinimumAvailability in a few places throughout Radarr.
+type Availability string
+
+// Availability / MinimumAvailability constants.
+// https://radarr.video/docs/api/#/MovieEditor/put_api_v3_movie_editor
+const (
+	AvailabilityToBeAnnounced Availability = "tba"
+	AvailabilityAnnounced     Availability = "announced"
+	AvailabilityInCinemas     Availability = "inCinemas"
+	AvailabilityReleased      Availability = "released"
+	AvailabilityDeleted       Availability = "deleted"
+)
+
+// Ptr returns a pointer to a minimum availability. Useful for a BulkEdit struct.
+func (a Availability) Ptr() *Availability {
+	return &a
 }
```

Comment:
I don't get the use case of this, can't you just use the `&` operator? why is a method needed?

Reply:
You cannot make pointers to constants, so the user would have to do:
```
  avail := radarr.AvailabilityInCinemas
```
Then pass in `&avail`. Having the `Ptr` method allows passing a value in 1 line.

Reply:
I showed how it works in the test.